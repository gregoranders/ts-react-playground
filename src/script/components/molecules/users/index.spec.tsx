import React, { create, mount } from '@app/test-utils';

import { DefaultUser } from '@models/user';

import * as TestSubject from '@molecules/users';

describe(`${TestSubject.UsersView.displayName}`, () => {
  it('exports default', () => {
    expect(TestSubject.default).toBeDefined();
  });

  const TestComponent = TestSubject.default;

  it('snapshot', (): void => {
    const snapshot = create(<TestComponent users={[DefaultUser]} />);
    snapshot.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('mount', () => {
    const testSubject = mount(<TestComponent users={[DefaultUser]} />);
    expect(testSubject.text()).toBe('First NameLast NameeMail');
  });
});
