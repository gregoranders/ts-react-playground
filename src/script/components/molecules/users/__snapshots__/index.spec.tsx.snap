// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersView snapshot 1`] = `
<dl>
  <dt>
    First Name
  </dt>
  <dd>
    
  </dd>
  <dt>
    Last Name
  </dt>
  <dd>
    
  </dd>
  <dt>
    eMail
  </dt>
  <dd>
    
  </dd>
</dl>
`;
